File: ios/BlakJaks/Shared/Components/TierBadge.swift
import SwiftUI

// MARK: - TierBadge
// Displays member tier with appropriate color.
// Standard (gray) / VIP (blue) / High Roller (purple) / Whale (gold)

struct TierBadge: View {
    let tier: String

    var tierColor: Color {
        switch tier.lowercased() {
        case "vip":          return .tierVIP
        case "high_roller", "high roller": return .tierHighRoller
        case "whale":        return .tierWhale
        default:             return .tierStandard
        }
    }

    var tierLabel: String {
        switch tier.lowercased() {
        case "high_roller": return "High Roller"
        default: return tier.capitalized
        }
    }

    var body: some View {
        Text(tierLabel)
            .font(.caption.weight(.semibold))
            .foregroundColor(tier.lowercased() == "whale" ? .black : .white)
            .padding(.horizontal, Spacing.sm)
            .padding(.vertical, Spacing.xs)
            .background(tierColor)
            .cornerRadius(6)
    }
}

// MARK: - Preview

#Preview {
    HStack(spacing: 12) {
        TierBadge(tier: "Standard")
        TierBadge(tier: "VIP")
        TierBadge(tier: "High Roller")
        TierBadge(tier: "Whale")
    }
    .padding()
    .background(Color.backgroundPrimary)
}

File: android/app/src/main/java/com/blakjaks/app/core/components/TierBadge.kt
package com.blakjaks.app.core.components

import androidx.compose.foundation.background
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.blakjaks.app.core.theme.*

@Composable
fun TierBadge(tier: String, modifier: Modifier = Modifier) {
    val color = when (tier.lowercase()) {
        "vip" -> TierVIP
        "high_roller", "high roller" -> TierHighRoller
        "whale" -> TierWhale
        else -> TierStandard
    }
    Text(
        text = tier.uppercase(),
        fontSize = 10.sp,
        fontWeight = FontWeight.Bold,
        color = color,
        modifier = modifier
            .background(color.copy(alpha = 0.15f), RoundedCornerShape(4.dp))
            .padding(horizontal = 8.dp, vertical = 3.dp)
    )
}
